pkg/sql/colexec/colexecbase/colexecbase_test: TestRandomizedCast failed

pkg/sql/colexec/colexecbase/colexecbase_test.TestRandomizedCast [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9073447?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9073447?buildTab=artifacts#/) on master @ [9b43dc50d468171bc9919dc684b7274b63e13da2](https://github.com/cockroachdb/cockroach/commits/9b43dc50d468171bc9919dc684b7274b63e13da2):


```
=== RUN   TestRandomizedCast
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/e8ecb0a2d1daf60d35be0846c86e1f60/logTestRandomizedCast3731116985
    test_log_scope.go:79: use -show-logs to present logs inline
    cast_test.go:88: unexpected error from the vectorized engine: runtime error: index out of range [3185] with length 2856
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e8ecb0a2d1daf60d35be0846c86e1f60/logTestRandomizedCast3731116985
--- FAIL: TestRandomizedCast (3.67s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomizedCast.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25426

--- a/pkg/sql/colexec/colexecbase/cast.eg.go
+++ b/pkg/sql/colexec/colexecbase/cast.eg.go
@@ -1246,7 +1246,7 @@ var _ colexecop.ClosableOperator = &castIdentityOp{}
 var identityOrder []int
 
 func init() {
-	identityOrder = make([]int, coldata.BatchSize())
+	identityOrder = make([]int, coldata.MaxBatchSize)
 	for i := range identityOrder {
 		identityOrder[i] = i
 	}

--- a/pkg/sql/colexec/colexecbase/cast_tmpl.go
+++ b/pkg/sql/colexec/colexecbase/cast_tmpl.go
@@ -314,7 +314,7 @@ var _ colexecop.ClosableOperator = &castIdentityOp{}
 var identityOrder []int
 
 func init() {
-	identityOrder = make([]int, coldata.BatchSize())
+	identityOrder = make([]int, coldata.MaxBatchSize)
 	for i := range identityOrder {
 		identityOrder[i] = i
 	}

